test/html/pipeline/mention_filter_test.rb

```diff
@@ -72,6 +72,34 @@ def test_links_to_more_info_when_info_url_given
       filter(body, '/', 'https://github.com/blog/821').to_html
   end
 
+  def test_base_url_slash
+    body = "<p>Hi, @jch!</p>"
+    link = "<a href=\"/jch\" class=\"user-mention\">@jch</a>"
+    assert_equal "<p>Hi, #{link}!</p>",
+      filter(body, '/').to_html
+  end
+
+  def test_base_url_under_custom_route
+    body = "<p>Hi, @jch!</p>"
+    link = "<a href=\"/userprofile/jch\" class=\"user-mention\">@jch</a>"
+    assert_equal "<p>Hi, #{link}!</p>",
+      filter(body, '/userprofile').to_html
+  end
+
+  def test_base_url_tilde
+    body = "<p>Hi, @jch!</p>"
+    link = "<a href=\"/~jch\" class=\"user-mention\">@jch</a>"
+    assert_equal "<p>Hi, #{link}!</p>",
+      filter(body, '~').to_html
+  end
```

Comment:
OK! I think it is reasonable not to guard for bad input. Lesson learned, thank you again! I have applied your commit. :beers: